### ISSUE TITLE
Use of `--main` as both a flag and an option leads to ambiguity

### ISSUE THREAD

**Issue by cdparks** [purescript/purescript#1204]
Assume we're trying to compile a module called Hello:

```purescript
module Hello where

import Debug.Trace

main = trace "Hello world!"
```

The following command succeeds:

```text
$ psc Hello.purs --output Hello.js --main=Hello
```

But the following command fails:

```text
$ psc Hello.purs --output Hello.js --main Hello
Error:
main is undefined
```

This is because `--main` with no trailing equals-sign gets parsed as a flag indicating that psc should look for main in the default module Main (which in this case does not exist).

The option parser for `--main` is in `psc/Main.hs - runMain`. Swapping the order of `noArgs <|> withArgs` in `runMain` makes both of the above commands succeed, but then `--main` cannot be used as a standalone flag. This may be a limitation of optparse-applicative. I haven't been able to find any examples of other projects using overlapping names for a flag and an option.

**Comment by paf31** [comment]
This has just been fixed and will be in 0.7, which I'll release very soon.

**Comment by cdparks** [comment]
Thanks!
